routine_butler/elements/routines_sidebar.py

```diff
@@ -481,25 +475,25 @@ def __init__(
 #                         target_duration_toggle.value
 #                     ),
 #                 )
-#             # bottom buttons
-#             ui.separator()
-#             with ui.row().classes(DFLT_ROW_CLASSES + f" pb-{V_SPACE}"):
-#                 # start routine
-#                 self.start_button = ui.button().classes("w-3/4")
-#                 self.start_button.props("icon=play_arrow color=positive")
-#                 # delete routine
-#                 self.delete_button = ui.button().classes("w-1/5")
-#                 self.delete_button.props("icon=cancel color=negative")
-#                 self.delete_button.on("click", self.on_delete)
-
-    def on_title_update(self, new_title):  # TODO: DB update
+            # # bottom buttons
+            # ui.separator()
+            # with ui.row().classes(DFLT_ROW_CLASSES + f" pb-{V_SPACE}"):
+            #     # start routine
+            #     self.start_button = ui.button().classes("w-3/4")
+            #     self.start_button.props("icon=play_arrow color=positive")
+            #     # delete routine
+            #     self.delete_button = ui.button().classes("w-1/5")
+            #     self.delete_button.props("icon=cancel color=negative")
+            #     self.delete_button.on("click", self.on_delete)
+
+
+    def on_title_update(self, new_title):
         logger.debug(
-            f'Updating title of routine {self.routine.id} to "{new_title}"'
+            f'Updating title of routine {self.routine_id} to "{new_title}"'
         )
-        self.header_label.set_text(new_title)
-        self.routine.title = new_title
-        self.session.add(self.routine)
-        self.session.commit()
+        with self.ctl.session() as session:
+            self.header_label.set_text(new_title)
+            self.ctl.rename_routine(session, self.routine_id, new_title)
```

Comment:
This is an example on how the Controller would be used.

I guess controller could be responsible for creating the "Session" as well.
Need to try to see which one feels better.